src/main/java/duke/DateString.java

```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Represents dates used by the Deadline and Event classes. Internally contains a
+ * java.time.LocalDate object that outputs the date in a standardised pattern,
+ * but silently falls back to using a string if the input cannot be parsed.
+ */
+public class DateString {
+    private final String dateString;
+    private LocalDate localDate;
+
+    /**
+     * Constructor which takes in a string and attempts to parse it to create a LocalDate
+     * object. If the parsing is unsuccessful, silently defaults to using a string instead.
+     * @param str
```

Comment:
Leaving a line between the description and parameters section could improve readability. Not sure if this is intended as I notice you have left a blank line for the other javadocs.